### DISCUSSION TITLE
uasyncio server awrite documentation?

### DISCUSSION THREAD

**Discussion by GM-Script-Writer-62850** [micropython/micropython#15893] [General]
I came across this: https://github.com/orgs/micropython/discussions/10745

I have always seen and used `writer.write(my_data_string)` what is this asynchronous writer i see in that thread (`await writer.awrite(my_data_string)`)

**Comment by peterhinch** [comment]
See [this explanation](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md#63-using-the-stream-mechanism).

**Reply by GM-Script-Writer-62850** [reply]
Not seeing anything about this '_a_write()'

**Reply by peterhinch** [reply]
To quote from the above reference:
> Writing to a StreamWriter occurs in two stages. The synchronous .write method concatenates data for later transmission. The asynchronous .drain causes transmission. To avoid allocation call .drain after each call to .write.

**Reply by GM-Script-Writer-62850** [reply]
that was my understanding of how it works
```
writer.write()
await writer.drain()
```
but i was asking were this comes from
```
await writer.awrite() # what is this???
await writer.drain()
```

**Reply by peterhinch** [reply]
It's legacy code and is deprecated. From the source:
```py
# Legacy uasyncio compatibility

async def stream_awrite(self, buf, off=0, sz=-1):
    if off != 0 or sz != -1:
        buf = memoryview(buf)
        if sz == -1:
            sz = len(buf)
        buf = buf[off : off + sz]
    self.write(buf)
    await self.drain()


Stream.aclose = Stream.wait_closed
Stream.awrite = stream_awrite
```
As you can see, internally it does what is described above.

**Reply by GM-Script-Writer-62850** [reply]
thanks
